doc/source/configuration/ci-cd.rst

```diff
@@ -0,0 +1,179 @@
+Kayobe automation provides users with an easy process to perform tasks such as: overcloud service deploy, config-diff, tempest testing, and many more.
+With it being integrated into platforms such as GitHub or GitLab it builds a close relationship between the contents of the deployments kayobe configuration and what is currently deployed.
+This is because operations such as opening a pull request will trigger a config diff to be generated providing insight on what impact it might have on services or a tempest test that could be scheduled to run daily providing knowledge of faults earlier than before.
+
+Kayobe automation has been designed to be independent of any CI/CD platform with all tasks running inside of a purpose built Kayobe container.
+However, platform specific workflows need to be deployed to bridge the gap between the contents of Kayobe Automation and these CI/CD platforms.
+Workflows are templated for each Kayobe configuration repository, ensuring appropriate workflow input parameters are defined, and any necessary customisations can be applied.
+The templating of workflows is offered through the `stackhpc.kayobe_workflows <https://github.com/stackhpc/ansible-collection-kayobe-workflows/>`__ collection which currently supports GitHub workflows.
+
+Runners
+-------
+
+Runners are purpose built services tied to a particular service vendor such as GitHub Actions or GitLab CI.
+These services will listen for jobs which have been tagged appropriately and dispatched to these specific runners.
+The runners will need to be deployed using existing roles and playbooks whereby the binary/package is downloaded and registered using a special token.
+
+GitHub Actions
+=================
+
+To enable CI/CD where GitHub Actions is used please follow the steps described below starting with the deployment of the runners.
+
+Runner Deployment
+-----------------
+
+1. Identify a suitable host for hosting the runners.
+    GitHub runners need to be deployed on a host which has not had Docker deployed using kolla.
+    This is because GitHub runners cannot provide `network options when running in a container <https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idcontaineroptions>`__.
+
+    Ideally an Infra VM could be used here or failing that the control host.
+    Wherever it is deployed the host will need access to the :code:`admin_network`, :code:`public_network` and the :code:`pulp registry` on the seed.
+
+2. Edit the environment's :code:`${KAYOBE_CONFIG_PATH}/environments/${KAYOBE_ENVIRONMENT}/inventory/groups` to add the predefined :code:`github-runners` group to :code:`infra-vms`
+
+.. code-block:: ini
+
+    [infra-vms:children]
+    github-runners
+
+3. Edit the environment's :code:`${KAYOBE_CONFIG_PATH}/environments/${KAYOBE_ENVIRONMENT}/inventory/hosts` to define the host(s) that will host the runners.
+
+.. code-block:: ini
+
+    [github-runners]
+    runner-01
```

Comment:
nit: It's a bit confusing that this almost matches the keys in the github_runners dict.

Reply:
Hopefully add `prod-` should help distinguish between host and runner. Or maybe `prod-runner-host-01`